ncm-metaconfig/src/main/perl/metaconfig.pm

```diff
@@ -123,7 +124,13 @@ sub handle_service
         return;
     }
 
-    $self->prepare_action($srv, $file, $actions) if ($fh->close());
+    my $changes = $fh->close();
+    if ( $changes > 0 ) {
```

Comment:
i reread `LC::Check::file`, there's no way that can return a negative number. on failure, it throws an error and returns undef (which is not handled in `CAF::FileWriter`).